storagemarket/provider.go

```diff
@@ -142,6 +148,14 @@ func (p *Provider) ExecuteDeal(dp *types.ClientDealParams) (pi *api.ProviderDeal
 		Transfer:           dp.Transfer,
 	}
 
+	// cleanup all state allocated here if deal is NOT accepted.
+	defer func() {
+		if pi != nil || err != nil {
+			_ = os.Remove(ds.InboundFilePath)
+			p.cleanupDealHandler(ds.DealUuid)
+		}
+	}()
+
```

Comment:
Why not put this block after the InboundFilePath and deal handler have been created?
It's a little tricky to follow where the return variables get set. I'd suggest trying to restructure the code so that you don't have to depend on reading the value of return variables in a defer statement.